include/core/surface_tension_model.h

```diff
@@ -231,10 +232,161 @@ class SurfaceTensionLinear : public SurfaceTensionModel
       std::fill(jacobian_vector.begin(), jacobian_vector.end(), 0);
   }
 
-private:
+protected:
   const double surface_tension_coefficient;
   const double T_0;
   const double surface_tension_gradient;
 };
 
+/**
+ * @brief Linear surface tension with phase change. The surface tension is given by:
+ * alpha_l*(sigma_0 + dsimga/dT * (T-T_0)).
+ */
+class SurfaceTensionPhaseChange : public SurfaceTensionLinear
+{
+public:
+  /**
+   * @brief Default constructor
```

Comment:
I would add the @param flags also to explain what are the parameters taken as inputs.